src/pipe.js

```diff
@@ -205,6 +205,18 @@
         };
     }
 
+    /**
+     * Preserve and execute the functions in the order
+     * the callbacks were registered
+     */
+    function onDoneHook(cb) {
+        var prevCb = hooks['onDone'];
+        hooks['onDone'] = function() {
+            prevCb();
```

Comment:
try-catch?

Reply:
It should blew up. Its better to not measure anything and throw instead of handling error and moving on with wrong measurements.

Reply:
I'd assume `done` event to be more generic to be honest as a consumer. Using it for perf measurements (what we do) is only one use-case (though, a real one!).

Why I asked, cause with normal `events` in node that'd be the case — chain doesn't break when there's an error in one of the handlers.

Reply:
Mb adding one more sentence https://github.com/zalando/tailor/blob/master/docs/hooks.md here would resolve potential issues?

Reply:
Done.